arm-runtimes/CMakeLists.txt

```diff
@@ -0,0 +1,780 @@
+    # architecture in the triple, because compiler-rt's cmake
+    # system doesn't recognize every possible Arm architecture
+    # version. So mostly we just say 'arm' and control the arch
+    # version via -march=armv7m (or whatever).
+    # Exceptions are architectures pre-armv7, which compiler-rt expects to
+    # see in the triple because that's where it looks to decide whether to
+    # use specific assembly sources.
+    if(TARGET_ARCH MATCHES "^armv[4-6]")
+        set(target_triple "${TARGET_ARCH}-none-eabi")
+    else()
+        set(target_triple "arm-none-eabi")
+    endif()
+    if(flags MATCHES "-mfloat-abi=hard")
+        # Also, compiler-rt looks in the ABI component of the
+        # triple to decide whether to use the hard float ABI.
+        set(target_triple "${target_triple}hf")
+    endif()
+    set(cpu_family arm)
+endif()
+
+# If any testing is enabled, prepare test executor settings.
+if (ENABLE_LIBC_TESTS OR ENABLE_COMPILER_RT_TESTS OR ENABLE_LIBCXX_TESTS)
+    if(TEST_EXECUTOR STREQUAL qemu)
+        if(TARGET_ARCH MATCHES "^aarch64")
+            find_program(QEMU_EXECUTABLE qemu-system-aarch64)
+        else()
+            find_program(QEMU_EXECUTABLE qemu-system-arm)
+        endif()
+
+        # Use colon as a separator because comma and semicolon are used for
+        # other purposes in CMake.
+        string(REPLACE " " ":" qemu_params_list "${QEMU_PARAMS}")
+
+        set(test_executor_params --qemu-command ${QEMU_EXECUTABLE} --qemu-machine ${QEMU_MACHINE})
+        if(QEMU_CPU)
+            list(APPEND test_executor_params --qemu-cpu ${QEMU_CPU})
+        endif()
+        if(qemu_params_list)
+            list(APPEND test_executor_params "--qemu-params=${qemu_params_list}")
+        endif()
+        set(
+            lit_test_executor
+            ${CMAKE_CURRENT_SOURCE_DIR}/test-support/lit-exec-qemu.py
+            ${test_executor_params}
+        )
+    elseif(TEST_EXECUTOR STREQUAL fvp)
+        if(NOT EXISTS "${FVP_INSTALL_DIR}")
+            message(FATAL_ERROR "FVPs must be installed to run tests using FVPs.")
+        endif()
+        set(
+            test_executor_params
+            --fvp-install-dir ${FVP_INSTALL_DIR}
+            --fvp-config-dir ${FVP_CONFIG_DIR}
+            --fvp-model ${FVP_MODEL}
+        )
+        string(REPLACE " " ";" fvp_config_list ${FVP_CONFIG})
+        foreach(cfg ${fvp_config_list})
+            set(
+                test_executor_params
+                ${test_executor_params}
+                --fvp-config ${cfg}
+            )
+        endforeach()
+        set(
+            lit_test_executor
+            ${CMAKE_CURRENT_SOURCE_DIR}/test-support/lit-exec-fvp.py
+            ${test_executor_params}
+        )
+    endif()
+    list(JOIN lit_test_executor " " lit_test_executor)
+endif()
+
+set(compile_arch_flags "--target=${target_triple} ${COMPILE_FLAGS}")
+set(link_flags "-nostartfiles -lcrt0-semihost -lsemihost -T picolibcpp.ld")
```

Comment:
Probably would be better named `test_link_flags` since it only gets used for linking compiler-rt and libcxx tests. It would sit better if moved slightly above into the `if` block that is only enabled with tests being enabled.
At the moment testing only works with picolibc, so a lot of the existing code setting up the tests was only written with picolibc in mind and will need changing for other libraries. For now attempting to enable tests will simply result in a configuration error.